firewood/src/shale/mod.rs

```diff
@@ -341,32 +338,33 @@ impl<T: Storable> StoredView<T> {
         &mut **self.mem
     }
 
-    fn estimate_mem_image(&self) -> Option<u64> {
-        let len = self.decoded.serialized_len();
+    /// Returns the serialized length of the item if it's less than the limit, otherwise `None`.
+    fn serialized_len(&self) -> Option<u64> {
+        let len = self.item.serialized_len();
         if len > self.len_limit {
             None
         } else {
             Some(len)
         }
     }
 
-    fn write_mem_image(&self, mem_image: &mut [u8]) -> Result<(), ShaleError> {
-        self.decoded.serialize(mem_image)
+    fn serialize(&self, mem_image: &mut [u8]) -> Result<(), ShaleError> {
+        self.item.serialize(mem_image)
     }
 
-    fn write(&mut self) -> &mut T {
-        &mut self.decoded
+    fn mut_item_ref(&mut self) -> &mut T {
```

Comment:
Again, `write` made it sound like we were actually writing something to a buffer but we actually weren't